### PR TITLE
add people to team

### PR DESCRIPTION
**Pull request by YrPryhoda**
* find users from mount of 'people'
* add from 1 to 4 people simultaneously

### CODE REVIEW

File: server/src/repositories/teams.repository.ts [hunk 1]
```diff
@@ -83,4 +84,36 @@ export class TeamRepository extends Repository<Team> {
...
+			}
+			// eslint-disable-next-line array-callback-return
+			users.map((user: UserProfile): void => {
+				const isExist = team.users?.find((el) => el.id === user.id);
+				if (!isExist) {
+					team?.users?.push(user);
+				}
+			});
+
+			await this.save(team);
```
Comment (kostia1359) [inline server/src/repositories/teams.repository.ts:107]: I suppose, this should return updated team. You don't need to do select query again

File: client/src/services/team.service.ts [hunk 2]
```diff
@@ -80,3 +82,27 @@ export const fetchTeams = async () => {
...
+	const result = await callWebApi({
+		method: 'POST',
+		endpoint: 'user/search',
+		body: {
+			id,
+			match
+		}
+	})
+	return await (result.json()) as WebApi.Entities.UserProfile[];
+}
```
Comment (kostia1359) [inline client/src/services/team.service.ts:96]: It looks strange, when you are searching data using POST method.
You can use GET method and query parameters

File: client/src/containers/TeamPage/logic/saga.ts [hunk 3]
```diff
@@ -66,6 +70,38 @@ export function* updateField(props: Props) {
 	}
 }
 
+export function* searchPeople(props: Props) {
+	try {
+		yield put(actions.searchPeopleLoader());
+		const users = yield call(findUsersColleagues, props.id, props.match);
+		const searchResult: { data: WebApi.Entities.UserProfile; key: string; title?: string; }[] = [];
+		users.filter((el: WebApi.Entities.UserProfile) => {
```
Comment (kostia1359) [inline client/src/containers/TeamPage/logic/saga.ts:78]: You shouldn't do search in saga. It is better to filter users in service
Also it is better to use forEach 
```suggestion
		users.forEach((el: WebApi.Entities.UserProfile) => {
```

File: client/src/containers/SearchPeopleAndTeamField/logic/saga.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	yield put(actions.SuccessLoading({ teams, people }));
+}
+*/
+export function* watchStartLoading() {
+	//	yield takeEvery(actionTypes.START_LOADING, fetchPeopleSearchPage);
+}
+
+export default function* projectsSaga() {
+	yield all([watchStartLoading()]);
+}
```
Comment (kostia1359) [inline client/src/containers/SearchPeopleAndTeamField/logic/saga.ts:17]: this saga shouldn't work at all)

**Review by kostia1359 — CHANGES_REQUESTED** [review]

File: server/src/repositories/teams.repository.ts [hunk 5]
```diff
@@ -83,4 +84,36 @@ export class TeamRepository extends Repository<Team> {
...
+			const team = await this.createQueryBuilder('Team')
+				.where('Team.id = :id', { id })
+				.leftJoinAndSelect('Team.users', 'Users')
+				.getOne();
+
+			if (!team) {
+				throw new Error('Team was not found');
+			}
+			// eslint-disable-next-line array-callback-return
+			users.map((user: UserProfile): void => {
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply server/src/repositories/teams.repository.ts:100]: good advice )

File: server/src/repositories/teams.repository.ts [hunk 6]
```diff
@@ -83,4 +84,36 @@ export class TeamRepository extends Repository<Team> {
 	deleteOneById(id: string) {
 		return this.delete(id);
 	}
+
+	// eslint-disable-next-line consistent-return
+	async addPeopleToTeam(id: string, users: UserProfile[]) {
+		try {
+			const team = await this.createQueryBuilder('Team')
+				.where('Team.id = :id', { id })
+				.leftJoinAndSelect('Team.users', 'Users')
```
[the remark this answers is not in the document]
Comment (YrPryhoda) [reply server/src/repositories/teams.repository.ts:93]: it just for validation and it won't be sent on frontend, so i suppose it isn't necessary

File: server/src/repositories/teams.repository.ts (hunk 1 above)
Comment (YrPryhoda) [reply server/src/repositories/teams.repository.ts:107]: yeah. you're rigth

File: client/src/containers/TeamPage/logic/saga.ts (hunk 3 above)
Comment (YrPryhoda) [reply client/src/containers/TeamPage/logic/saga.ts:78]: let it been as you wish

File: client/src/containers/SearchPeopleAndTeamField/logic/saga.ts (hunk 4 above)
Comment (YrPryhoda) [reply client/src/containers/SearchPeopleAndTeamField/logic/saga.ts:17]: i've deleted al "logic" repository

**Review by kostia1359 — APPROVED** [review]
LGTM!

**Pull request merged**